lua/freeze-code/commands.lua

```diff
@@ -121,7 +121,7 @@ local copy_by_os = function(opts)
     if vim.env.XDG_SESSION_TYPE == "x11" then
       cmd = { "xclip", "-selection", "clipboard", "-t", "image/png", "-i", filename }
     else
-      cmd = { "wl-copy", "<", filename }
+      cmd = { "sh", "-c", "wl-copy <" .. filename }
```

Comment:
@chenrry666 does it work the intended way if instead of `{ "sh", "-c", "wl-copy <" .. filename }`, you use `{ "sh", "-c", "wl-copy", "<", filename }`?

Reply:
It doesn't work. In a word, the goal is to make the string `wl-copy < file` a single argument ( a string with spaces in it ), instead of multiple arguments. 

BTW, why do we wrap it using `sh -c`:
> If {cmd} is a List it runs directly (no 'shell').
> If {cmd} is a String it runs in the 'shell'

Since we pass a list as an argument, it directly runs, so shell syntax (`<` for redirecting standard input ) won't work. 

Ref: [":help system"](https://neovim.io/doc/user/builtin.html#system())

Reply:
> BTW, why do we wrap it using `sh -c`:

This one is not in the codebase, this is something you added in the changes.

> It doesn't work. In a word, the goal is to make the string `wl-copy < file` a single argument ( a string with spaces in it ), instead of multiple arguments.

`vim.fn.system({ "wl-copy", "<", filename })` would be the same as `bash -c 'wl-copy < freeze.png'`

> [!note]
> Using `freeze.png` as default filename, which is being expanded with the `dir` option in the config, being `dir = vim.env.PWD` as default.
>
> https://github.com/AlejandroSuero/freeze-code.nvim/blob/90291a4dde597cacdfc6cd56d03fd781acd2e2f1/lua/freeze-code/binary/binary_handler.lua#L36-L49

If `$ wl-copy < foobar.png` or `$ wl-copy < path/to/foobar.png` only copies as string instead of image, maybe there is an flag with `wl-copy` that ensures copy a filetype like `xclip -t image/png -i foobar.png` does (I don't have a wayland machine so I'm not familiar with how wayland does it.

On the other hand, you can try changing it to `cmd = { "wl-copy", "<", filename }` how it was and then when returning `vim.fn.system(table.concat(cmd, " ")`, but one leads to MacOS and maybe Windows failing to copy it, so if it works better `table.concat` after `cmd = { ... }` inside the if statement.

Reply:
Under the [wl-copy manpage](https://manpages.debian.org/buster/wl-clipboard/wl-copy.1.en.html) it shows at the bottom part how to copy an image, it also indicates that `-t` can be use for `mime/type` so it can be use in that way.

Reply:
Specifying the -t or --type argument with wl-copy is also not working:
![image](https://github.com/user-attachments/assets/a0f4d233-1044-45ef-9306-b6dbbb7eb05c)

I had to do below change in order to make it working:
![image](https://github.com/user-attachments/assets/39a68711-8452-439b-83a5-0303236418c5)

As per ChatGPT, it has the issue with redirection, please read:
![image](https://github.com/user-attachments/assets/c1ec4f4d-cf99-4a5d-98f9-3f897b23eaf6)

To solve this we can modify the code and explicitely check for wl-copy case similar to gclip:
![image](https://github.com/user-attachments/assets/812fc901-feb2-4b4c-87a2-5de75a164a9c)

Reply:
The it's settle, I tried doing `ls > dirlist` and seems like it doesn't work either.